app/controllers/feedbacks_controller.rb

```diff
@@ -24,7 +25,7 @@ def create
     @feedback.feedbackable = @feedbackable
     authorize @feedback
     if @feedback.save
-      redirect_to @feedback.volunteer, make_notice
+      redirect_to create_redirect, make_notice
```

Comment:
And for this you could simply pass a get param to the 
`/volunteers/99/assignments/99/feedbacks/new?last_submitted=true`
in the `/app/views/feedbacks/_form.html.slim` you could then add this get parameter to the action if existing, and then your controller `create` action will know where its comming from - where to redirect to. Or at least without relying on refferers.